src/admin-copy-user-data.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-await-in-loop,no-console */
+/**
+ * TPM ONLY SCRIPT! BE CAREFUL COPYING TO PROD!
+ * Script to copy data from one user on production or dev to another user on dev.
+ * Requires service accounts for database.
+ * serviceAccount.json (if using dev) and serviceAccountProd.json (if using prod) must be at the root.
+ *
+ * From root, run: `npm run ts-node -- src/admin-copy-user-data.ts -f <FROM_ENV>/<FROM_USER> -t <TO_ENV>/<TO_USER> -e <EXECUTE>`
+ * FROM_ENV and TO_ENV should be either "dev" or "prod"
+ * EXECUTE is an optional argument that is false by default. It can be only `true` or `false`.
+ *    EXECUTE=false will preview the changes, EXECUTE=true will do the changes
+ * EXAMPLE: `npm run ts-node -- src/admin-copy-user-data.ts -f prod/noschiff.dev@gmail.com -t dev/nps39@cornell.edu -e true`
+ */
+
+import { cert, initializeApp } from 'firebase-admin/app';
+import { getFirestore } from 'firebase-admin/firestore';
+import parseArgs from 'minimist';
+
+const collections = [
+  'user-name',
+  'user-semesters',
+  'user-toggleable-requirement-choices',
+  'user-overridden-fulfillment-choices',
+  'user-selectable-requirement-choices',
+  'user-subject-colors',
+  'user-unique-incrementer',
+  'user-onboarding-data',
+];
+
+const args = parseArgs(process.argv, {
+  string: ['f', 't'],
+  boolean: ['e'],
+  alias: { from: 'f', to: 't', execute: 'e' },
+  default: { e: false },
+});
+
+if (args.from && args.to && 'execute' in args) {
+  const [FROM_ENV, FROM] = args.from.split('/');
+  const [TO_ENV, TO] = args.to.split('/');
+  if (
+    (args.execute === true || args.execute === false) &&
+    (FROM_ENV === 'prod' || FROM_ENV === 'dev') &&
+    (TO_ENV === 'prod' || TO_ENV === 'dev') &&
+    FROM &&
+    TO
+  ) {
+    execute({
+      fromUser: FROM,
+      fromEnv: FROM_ENV,
+      toUser: TO,
+      toEnv: TO_ENV,
+      execute: args.execute,
+    }).then(copied => {
+      if (args.execute) console.log(`Copied: [${copied}] from ${args.from} to ${args.to}`);
+      else console.log(copied);
+    });
+  } else {
+    throw new Error('Refer to the documentation to correctly run this script.');
+  }
+} else {
+  throw new Error('Please include all required command line arguments.');
+}
+
+async function execute(
+  options: Readonly<{
+    fromUser: string;
+    toUser: string;
+    fromEnv: 'prod' | 'dev';
+    toEnv: 'prod' | 'dev';
+    execute: boolean;
+  }>
+): Promise<string[]> {
+  let fromDb;
+  let toDb;
+  if (options.fromEnv === 'dev' || options.toEnv === 'dev') {
+    const dev = initializeApp({
+      credential: cert('serviceAccount.json'),
```

Comment:
It works fine for me when the service account is at the root. Ideally, we'd store the credential in an environment variable like how we do it on automatic deployments, but I figured just using a JSON would be the easiest. Also, this `cert` can be weird with what it requires. I think we can actually read the JSON in using some other method and then pass that object to the cert if we prefer, but it can also go find it itself.